### ISSUE TITLE
Problem when I install ruby

### ISSUE THREAD

**Issue by visualcode** [rvm/rvm#3669]
Hello guys,
I'm stuck in this error when I install any ruby versions:

Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.11/x86_64/ruby-2.3.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates in '/usr/local/etc/openssl/cert.pem' are already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/emanuelelivi/.rvm/rubies/ruby-2.3.0, this may take a while depending on your cpu(s)...
ruby-2.3.0 - #downloading ruby-2.3.0, this may take a while depending on your connection...
ruby-2.3.0 - #extracting ruby-2.3.0 to /Users/emanuelelivi/.rvm/src/ruby-2.3.0.-
ruby-2.3.0 - #configuring......................................................|
ruby-2.3.0 - #post-configuration.
ruby-2.3.0 - #compiling........................................................|
ruby-2.3.0 - #installing.............
Error running '__rvm_make install',
showing last 15 lines of /Users/emanuelelivi/.rvm/log/1461523799_ruby-2.3.0/install.log
	from ./tool/rbinstall.rb:686:in `block in <class:Installer>'
	from ./tool/rbinstall.rb:754:in `block (2 levels) in <main>'
	from /Users/emanuelelivi/.rvm/src/ruby-2.3.0/lib/rubygems/specification.rb:821:in `block in each_spec'
	from /Users/emanuelelivi/.rvm/src/ruby-2.3.0/lib/rubygems/specification.rb:743:in `block (2 levels) in each_gemspec'
	from /Users/emanuelelivi/.rvm/src/ruby-2.3.0/lib/rubygems/specification.rb:742:in `each'
	from /Users/emanuelelivi/.rvm/src/ruby-2.3.0/lib/rubygems/specification.rb:742:in `block in each_gemspec'
	from /Users/emanuelelivi/.rvm/src/ruby-2.3.0/lib/rubygems/specification.rb:741:in `each'
	from /Users/emanuelelivi/.rvm/src/ruby-2.3.0/lib/rubygems/specification.rb:741:in `each_gemspec'
	from /Users/emanuelelivi/.rvm/src/ruby-2.3.0/lib/rubygems/specification.rb:819:in `each_spec'
	from ./tool/rbinstall.rb:751:in `block in <main>'
	from ./tool/rbinstall.rb:801:in `block in <main>'
	from ./tool/rbinstall.rb:798:in `each'
	from ./tool/rbinstall.rb:798:in `<main>'
make: *** [do-install-nodoc] Error 1
++ return 2
There has been an error while running make install. Halting the installation.

basically this one is the error:
**There has been an error while running make install. Halting the installation.**

**Comment by visualcode** [comment]
Any ideas?

**Comment by pkuczynski** [comment]
We just uploaded 2.3.1 binaries for OSX El Capitan.

```
rvm get master
rvm install 2.3.1
```

should solve your problem.

**Comment by nettymaza** [comment]
@pkuczynski I am running into this issue on High Sierra. I tried your solution but still getting an error on the compiler step same as original post

```ruby-2.3.1 - #extracting ruby-2.3.1 to /Users/nettymaza/.rvm/src/ruby-2.3.1....
ruby-2.3.1 - #applying patch /Users/nettymaza/.rvm/patches/ruby/ruby_2_3_gcc7.patch.
ruby-2.3.1 - #applying patch /Users/nettymaza/.rvm/patches/ruby/2.3.1/random_c_using_NR_prefix.patch.
ruby-2.3.1 - #configuring...........................................................
ruby-2.3.1 - #post-configuration.
ruby-2.3.1 - #compiling...........
Error running '__rvm_make -j4',
please read /Users/nettymaza/.rvm/log/1512245316_ruby-2.3.1/make.log
There has been an error while running make. Halting the installation.```

Is this a known issue on High Sierra? Maybe a similar update to the binaries is needed?

**Comment by muhammadarsal** [comment]
Same issue installing version 1.9.3 on High Sierra.

`ruby-1.9.3-p551 - #removing src/ruby-1.9.3-p551..
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.13/x86_64/ruby-1.9.3-p551.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates bundle '/usr/local/etc/openssl/cert.pem' is already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/arsal/.rvm/rubies/ruby-1.9.3-p551, this may take a while depending on your cpu(s)...
ruby-1.9.3-p551 - #downloading ruby-1.9.3-p551, this may take a while depending on your connection...
ruby-1.9.3-p551 - #extracting ruby-1.9.3-p551 to /Users/arsal/.rvm/src/ruby-1.9.3-p551....
ruby-1.9.3-p551 - #applying patch /Users/arsal/.rvm/patches/ruby/GH-488.patch.
ruby-1.9.3-p551 - #applying patch /Users/arsal/.rvm/patches/ruby/1.9.3/CVE-2015-1855-p484.patch.
ruby-1.9.3-p551 - #applying patch /Users/arsal/.rvm/patches/ruby/1.9.3/update-autoconf.patch.
ruby-1.9.3-p551 - #applying patch /Users/arsal/.rvm/patches/ruby/1.9.3/openssl3.patch.
ruby-1.9.3-p551 - #configuring...............................................
ruby-1.9.3-p551 - #post-configuration.
ruby-1.9.3-p551 - #compiling........................................................................................................................................................-
Error running '__rvm_make -j 1',
please read /Users/arsal/.rvm/log/1513225777_ruby-1.9.3-p551/make.log
There has been an error while running make. Halting the installation.`

**Comment by anoobbava** [comment]
the same issue happened as Explained by @muhammadarsal  also

**Comment by anoobbava** [comment]
@muhammadarsal  Please try as explained in the below link
https://stackoverflow.com/questions/14114976/no-binary-rubies-available-for-osx-10-8-x86-64-ruby-1-9-2-p320

Results are
=========

Macs-MacBook-Pro:~ mac$ rvm use --install 1.9.3
Required ruby-1.9.3-p551 is not installed - installing.
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.13/x86_64/ruby-1.9.3-p551.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Installing requirements for osx.
Updating system......
Installing required packages: coreutils, gcc@6....
==> Upgrading 1 outdated package, with result:
automake 1.15.1 -> 1.16.1
==> Upgrading automake 
==> Downloading https://homebrew.bintray.com/bottles/automake-1.16.1.high_sierra.bottle.tar.gz
######################################################################## 100.0%
==> Pouring automake-1.16.1.high_sierra.bottle.tar.gz
🍺  /usr/local/Cellar/automake/1.16.1: 131 files, 3MB
Certificates bundle '/usr/local/etc/openssl/cert.pem' is already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/mac/.rvm/rubies/ruby-1.9.3-p551, this may take a while depending on your cpu(s)...
ruby-1.9.3-p551 - #downloading ruby-1.9.3-p551, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 9813k  100 9813k    0     0   597k      0  0:00:16  0:00:16 --:--:-- 1177k
ruby-1.9.3-p551 - #extracting ruby-1.9.3-p551 to /Users/mac/.rvm/src/ruby-1.9.3-p551.....
ruby-1.9.3-p551 - #applying patch /Users/mac/.rvm/patches/ruby/GH-488.patch.
ruby-1.9.3-p551 - #applying patch /Users/mac/.rvm/patches/ruby/1.9.3/CVE-2015-1855-p484.patch.
ruby-1.9.3-p551 - #applying patch /Users/mac/.rvm/patches/ruby/1.9.3/update-autoconf.patch.
ruby-1.9.3-p551 - #applying patch /Users/mac/.rvm/patches/ruby/1.9.3/openssl3.patch.
ruby-1.9.3-p551 - #configuring.
Error running './configure --prefix=/Users/mac/.rvm/rubies/ruby-1.9.3-p551 --with-opt-dir=/usr/local/opt/libyaml:/usr/local/opt/readline:/usr/local/opt/libksba:/usr/local/opt/openssl --without-tcl --without-tk --disable-install-doc --enable-shared',
please read /Users/mac/.rvm/log/1529595800_ruby-1.9.3-p551/configure.log
There has been an error while running configure. Halting the installation.
Gemset '' does not exist, 'rvm ruby-1.9.3-p551 do rvm gemset create ' first, or append '--create'.
Macs-MacBook-Pro:~ mac$ rvm ruby-1.9.3-p551
Required ruby-1.9.3-p551 is not installed.
To install do: 'rvm install "ruby-1.9.3-p551"'
Macs-MacBook-Pro:~ mac$ rvm install "ruby-1.9.3-p551"
ruby-1.9.3-p551 - #removing src/ruby-1.9.3-p551..
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.13/x86_64/ruby-1.9.3-p551.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates bundle '/usr/local/etc/openssl/cert.pem' is already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/mac/.rvm/rubies/ruby-1.9.3-p551, this may take a while depending on your cpu(s)...
ruby-1.9.3-p551 - #downloading ruby-1.9.3-p551, this may take a while depending on your connection...
ruby-1.9.3-p551 - #extracting ruby-1.9.3-p551 to /Users/mac/.rvm/src/ruby-1.9.3-p551.....
ruby-1.9.3-p551 - #applying patch /Users/mac/.rvm/patches/ruby/GH-488.patch.
ruby-1.9.3-p551 - #applying patch /Users/mac/.rvm/patches/ruby/1.9.3/CVE-2015-1855-p484.patch.
ruby-1.9.3-p551 - #applying patch /Users/mac/.rvm/patches/ruby/1.9.3/update-autoconf.patch.
ruby-1.9.3-p551 - #applying patch /Users/mac/.rvm/patches/ruby/1.9.3/openssl3.patch.
ruby-1.9.3-p551 - #configuring...............................................
ruby-1.9.3-p551 - #post-configuration.
ruby-1.9.3-p551 - #compiling................................................^[[B^[[B^R
^[[B^[......................................................................................................................|
ruby-1.9.3-p551 - #installing...........
ruby-1.9.3-p551 - #making binaries executable..
ruby-1.9.3-p551 - #downloading rubygems-2.7.7
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  894k  100  894k    0     0   850k      0  0:00:01  0:00:01 --:--:--  850k
No checksum for downloaded archive, recording checksum in user configuration.
ruby-1.9.3-p551 - #extracting rubygems-2.7.7.....
ruby-1.9.3-p551 - #removing old rubygems........
ruby-1.9.3-p551 - #installing rubygems-2.7.7................................
ruby-1.9.3-p551 - #gemset created /Users/mac/.rvm/gems/ruby-1.9.3-p551@global
ruby-1.9.3-p551 - #importing gemset /Users/mac/.rvm/gemsets/global.gems.............................................
ruby-1.9.3-p551 - #generating global wrappers.......
ruby-1.9.3-p551 - #gemset created /Users/mac/.rvm/gems/ruby-1.9.3-p551
ruby-1.9.3-p551 - #importing gemsetfile /Users/mac/.rvm/gemsets/default.gems evaluated to empty gem list
ruby-1.9.3-p551 - #generating default wrappers.......
ruby-1.9.3-p551 - #adjusting #shebangs for (gem irb erb ri rdoc testrb rake).
Install of ruby-1.9.3-p551 - #complete 
WARNING: Please be aware that you just installed a ruby that is no longer maintained (2017-04-01), for a list of maintained rubies visit:

    http://bugs.ruby-lang.org/projects/ruby/wiki/ReleaseEngineering

Please consider upgrading to ruby-2.5.1 which will have all of the latest security patches.
Ruby was built without documentation, to build it run: rvm docs generate-ri
Macs-MacBook-Pro:~ mac$ 
Macs-MacBook-Pro:~ mac$ rvm list
   jruby-9.1.1.0 [ x86_64 ]
=> ruby-1.9.3-p551 [ x86_64 ]
   ruby-2.0.0-p648 [ x86_64 ]
   ruby-2.3.0 [ x86_64 ]
   ruby-2.3.1 [ x86_64 ]
   ruby-2.3.2 [ x86_64 ]
   ruby-2.4.0 [ x86_64 ]
 * ruby-2.4.1 [ x86_64 ]
   ruby-2.4.2 [ x86_64 ]
   ruby-2.5.0 [ x86_64 ]

# => - current
# =* - current && default
#  * - default

Macs-MacBook-Pro:~ mac$
